benchmarks/dynamo/distributed.py

```diff
@@ -0,0 +1,164 @@
+import argparse
+from functools import partial
+
+import numpy as np
+import tabulate
+import torch
+
+import torch._dynamo as dynamo
+import torch.multiprocessing as mp
+import torch.utils._pytree as pytree
+from torch._dynamo.testing import reduce_to_scalar_loss
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.profiler import profile, ProfilerActivity, record_function
+
+try:
+    from .common import timed
+    from .dist_util import apply_fsdp, cleanup, get_model, model_iter_fn, setup
+except ImportError:
+    from common import timed
+    from dist_util import apply_fsdp, cleanup, get_model, model_iter_fn, setup
+
+
+def profile_model(args, model, inputs, rank):
+    with profile(activities=[ProfilerActivity.CPU, ProfilerActivity.CUDA]) as prof:
+        for i in range(args.repeat):
+            with record_function("Forward"):
+                outputs = model(*inputs)
+                loss = reduce_to_scalar_loss(outputs)
+            with record_function("Backward"):
+                loss.backward()
+    if rank == 0:
+        prof.export_chrome_trace(args.trace_file)
+
+
+def run_model(args, model, inputs, rank, world_size, key, result_q):
+    setup(rank, world_size)
+    if args.device == "cuda":
+        # needed for FSDP
+        torch.cuda.set_device(rank)
```

Comment:
can do do rank % torch.cuda.device_count() ? This would allow this to work on more cases.